### ISSUE TITLE
Attribute Table Frame limits before sorting

### ISSUE THREAD

**Issue by ianturton** [qgis/QGIS#32901]
**Describe the bug**
When you apply a limit and a sort key to the Print Layout frame view it limits the length of the table before it sorts the features meaning the 1st feature may be missing from the table.

**How to Reproduce**
Add a layer and in the print layout add an attribute table, under attributes set a sort field and in feature filtering set a number of rows. As you increase and decrease the number of rows you will see the 1st feature change.

![bug](https://user-images.githubusercontent.com/122529/68995525-dc20d300-0886-11ea-8eee-6e535927c9f5.png)

In the above the only difference between the tables is the number of features in the filter.

Load the following project and datafile to reproduce

[bug_project.zip](https://github.com/qgis/QGIS/files/3854375/bug_project.zip)
[states.zip](https://github.com/qgis/QGIS/files/3854376/states.zip)

**QGIS and OS versions**

QGIS version
3.10.0-A Coruña
QGIS code revision
6c816b4204
Compiled against Qt
5.9.5
Running against Qt
5.9.5
Compiled against GDAL/OGR
2.2.3
Running against GDAL/OGR
2.2.3
Compiled against GEOS
3.6.2-CAPI-1.10.2
Running against GEOS
3.7.1-CAPI-1.11.1 27a5e771
Compiled against SQLite
3.22.0
Running against SQLite
3.22.0
PostgreSQL Client Version
10.10 (Ubuntu 10.10-0ubuntu0.18.04.1)
SpatiaLite Version
4.3.0a
QWT Version
6.1.3
QScintilla2 Version
2.10.2
PROJ.4 Version
493
OS Version
Ubuntu 18.04.3 LTS
Active python plugins
EasyCustomLabeling; 
vrtbuilderplugin; 
autoSaver; 
QGIS3-getWKT; 
qgis_hats; 
MemoryLayerSaver; 
isochrones; 
firstaid; 
OSMDownloader; 
qchainage; 
plugin_reloader; 
polystrip; 
save_attributes; 
cartogram3; 
Hqgis; 
pluginbuilder3; 
quick_map_services; 
pointsamplingtool; 
gml_application_schema_toolbox; 
DataPlotly; 
QuickWKT; 
timemanager; 
processing; 
MetaSearch; 
db_manager
**Additional context**

**Comment by gioman** [comment]
yeah...there is something fishy going on (tried on the latest 3.4 here).

**Comment by Pedro-Murteira** [comment]
@ianturton I can no longer replicate this bug on QGIS 3.16.14 and 3.22.1.
